.gitignore

```diff
@@ -74,6 +74,7 @@ Thumbs.db
 # Ignore Terasology output
 *.log
 /saves/
+/configs/
```

Comment:
This would have been better as a part of #3118, but I think that it should be fine here. It is related to FlexibleConfigs, and in my opinion the change is too small to warrant a separate PR.

Reply:
The location of this is quite relevant to my interests :-)

With `-homedir` enabled (default in IntelliJ) stuff does go to the root of the workspace, which counts as the data directory. So stuff like `logs` and `saves` go there as well. `configs` would be a new thing. I imagine if you ran without `-homedir` the `configs` would correctly end up under the user home default location for the data dir, but that's irrelevant to the `.gitignore`

As for what to name it and where to put it I've recently been thinking about a new thing to let us use some limited file access from modules. For instance a `motd.txt` file with a new MOTD module taking over from the engine. Rather than read that from a workspace / data dir root it could live in a deeper directory like `saves` to keep the root cleaner.

Whether config stuff should go under `saves` or a new thing like `configs` I'm not sure about. I was leaning toward introducing a proposal to have a new `sandbox` directory at the same level as `saves` to store module-specific things like config, temp files, and so on (the root of the subdir could perhaps store non-module specific stuff like FC sections). Also a spot to output prefabs created by the StructureTemplate toolbox, or new behavior tree files. Stuff in there would be installation-wide, any individual save world that wants its own unique files would be something else.

Point of using "sandbox" would be to convey the idea that files within it can be read from modules and relate mostly to those, while stuff in the root are very engine specific (existing `config.cfg` and stuff in `saves` relate to individual worlds (games, really) and are irrelevant to anything else. The content of the dir should be considered transient - it could be deleted and while you lose some customizations/config the game will run fine with defaults.

I should write this up in its own issue but spotted this by chance and wanted to at least mention it plus a quick summary as a heads-up :-)

Reply:
My opinion:
1. configs (notice plural) should go in a /configs folder at the same level with logs and saves, wherever those are according to the use (or not) of -homedir. At least for the time being. When/If we get to per-save and per-module configs we will have to expand this functionality accordingly, with configs in /saves and /configs/modules/MyModule.cfg (this is just an idea). 
2. configs should certainly not go under /saves unless they are per-save, which currently they are not.